sagemaker-pipelines/tabular/local-mode/sagemaker-pipelines-local-mode.ipynb

```diff
@@ -0,0 +1,1132 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.workflow.fail_step import FailStep\n",
+    "\n",
+    "step_fail = FailStep(\n",
+    "    name=\"AbaloneMSEFail\",\n",
+    "    error_message=Join(on=\" \", values=[\"Execution failed due to MSE >\", mse_threshold]),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define a Condition Step to Check Accuracy and Conditionally Create a Model and Run a Batch Transformation Or Terminate the Execution in Failed State\n",
+    "\n",
+    "In this step, the model is registered only if the accuracy of the model, as determined by the evaluation step `step_eval`, exceeded a specified value. Otherwise, the pipeline execution fails and terminates. A `ConditionStep` enables pipelines to support conditional execution in the pipeline DAG based on the conditions of the step properties.\n",
+    "\n",
+    "In the following section, you:\n",
+    "\n",
+    "* Define a `ConditionLessThanOrEqualTo` on the accuracy value found in the output of the evaluation step, `step_eval`.\n",
+    "* Use the condition in the list of conditions in a `ConditionStep`.\n",
+    "* Pass the `CreateModelStep` and `TransformStep` steps into the `if_steps` of the `ConditionStep`, which are only executed if the condition evaluates to `True`.\n",
+    "* Pass the `FailStep` step into the `else_steps`of the `ConditionStep`, which is only executed if the condition evaluates to `False`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.workflow.conditions import ConditionLessThanOrEqualTo\n",
+    "from sagemaker.workflow.condition_step import ConditionStep\n",
+    "from sagemaker.workflow.functions import JsonGet\n",
+    "\n",
+    "cond_lte = ConditionLessThanOrEqualTo(\n",
+    "    left=JsonGet(\n",
+    "        step_name=step_eval.name,\n",
+    "        property_file=evaluation_report,\n",
+    "        json_path=\"regression_metrics.mse.value\",\n",
+    "    ),\n",
+    "    right=mse_threshold,\n",
+    ")\n",
+    "\n",
+    "step_cond = ConditionStep(\n",
+    "    name=\"AbaloneMSECond\",\n",
+    "    conditions=[cond_lte],\n",
+    "    if_steps=[step_create_model, step_transform],\n",
+    "    else_steps=[step_fail],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define a Pipeline using `LocalPipelineSession`\n",
+    "\n",
+    "In this section, combine the steps into a Pipeline so it can be executed. We provide a `LocalPipelineSession` object to the `Pipeline` so that when executed, all the steps in the pipeline will run locally on the  machine. By switching the `LocalPipelineSession` to a `sagemaker.session.Session` object, you can switch the execution to run in the cloud on SageMaker instances."
```

Comment:
By switching the `LocalPipelineSession` to a `PipelineSession`?

Reply:
Done